llvm/lib/Target/DirectX/DXILOpLowering.cpp

```diff
@@ -770,8 +770,14 @@ class OpLowerer {
         continue;
       Intrinsic::ID ID = F.getIntrinsicID();
       switch (ID) {
-      default:
+      case Intrinsic::dx_resource_casthandle:
+      case Intrinsic::not_intrinsic:
         continue;
+      default: {
+        DiagnosticInfoUnsupported Diag(F, "Unknown intrinsic?");
+        M.getContext().diagnose(Diag);
```

Comment:
Should we also set HasErrors here? It doesn't actually matter, since we won't leave the casthandles in an inconsistent state and we don't key anything else off of that, but it seems weird not to

Reply:
I'll add it for consistency. We don't seem to be able to do multiple Diagnostic errors per test case. Ie I can only invoke `M.getContext().diagnose(...)` once. So setting HasErrors seemed pointless.

Reply:
That's because you're using `opt`, which has the default error handler that just exits the program. If you want a test case with multiple errors you need to use `llc` (or `clang`, but that wouldn't be appropriate for this kind of test)